samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -262,6 +275,14 @@ public void run(Duration timeout) {
     if (status.getStatusCode() == ApplicationStatus.StatusCode.UnsuccessfulFinish) {
       throw new SamzaException(ExceptionUtils.getStackTrace(status.getThrowable()));
     }
+    // Clean the state directory if any
+    Path stateDir = Paths.get(configs.get(JobConfig.JOB_NON_LOGGED_STORE_BASE_DIR()));
+    if (!Files.exists(stateDir)) return;
+    try {
+      Files.walk(stateDir).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+    } catch (IOException e) {
+      throw new SamzaException("Could not delete the JOB_NON_LOGGED_STORE_BASE_DIR due to: \n" + e.getMessage());
```

Comment:
This message has to be warn/error. It will be better to not fail the test for this. What do you think?